### PR TITLE
[[ Bug 22605 ]] Improve storyboard launch file support for dark mode

### PR DESCRIPTION
**Pull request by montegoulding**
This patch adds support to load different images and use the system background color in launch screens when supporting dark mode on iOS. This patch also fixes a couple of minor bugs.

### CODE REVIEW

File: ide-support/revsaveasiosstandalone.livecodescript [hunk 1]
```diff
@@ -938,53 +938,60 @@ private command revDecorateImageAssets pTarget, pSettings, pBaseFolder, pAppBund
...
-            case "3x"
-               put item 1 of tDimensions div 3, \
-                     item 2 of tDimensions div 3 into tDimensions
-               break
-            default
-            case "1x"
-               break
-         end switch
-      end if
+   repeat for each item tAppearance in "light,dark"
```
Comment (runrevmark) [inline ide-support/revsaveasiosstandalone.livecodescript:941]: @montegoulding You mentioned offline about the dark mode related stuff needing Xcode 11 tools and above. What happens if you use older tools - does this break, or do you just not get the dark mode settings? (The latter is fine, the former not!).

Comment (montegoulding) [reply ide-support/revsaveasiosstandalone.livecodescript:941]: @runrevmark it just doesn't show the dark mode version but still works fine. That's with Xcode 9.4 tools. I'm not sure what min Xcode version we support. @livecodepanos ?

Comment (runrevmark) [reply ide-support/revsaveasiosstandalone.livecodescript:941]: Oh that's absolutely fine - I think that's the earliest toolset we can use :)

**Review by runrevmark — APPROVED** [review]

**Comment by runrevmark** [comment]
@livecode-vulcan review ok 2b53d5c

**Comment by livecode-vulcan** [comment]
💙 review by @runrevmark ok 2b53d5cb9eba9c92fc06a7267a0bf2227f10be71

**Comment by livecode-vulcan** [comment]
😎 test success 2b53d5cb9eba9c92fc06a7267a0bf2227f10be71

* try-community-armv6-android-sdk26_ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-armv6-android-sdk26_ndk16r15/builds/784)
* try-community-armv7-android-ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-armv7-android-ndk16r15/builds/193)
* try-community-arm64-android-ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-arm64-android-ndk16r15/builds/193)
* try-community-x86-android-ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-x86-android-ndk16r15/builds/193)
* try-community-x86_64-android-ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-x86_64-android-ndk16r15/builds/193)
* try-community-js-emscripten-sdk1.35: [success](https://vulcan.livecode.com/builders/try-community-js-emscripten-sdk1.35/builds/2310)
* try-community-universal-ios-iphoneos13.4: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphoneos13.4/builds/43)
* try-community-universal-ios-iphonesimulator13.4: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphonesimulator13.4/builds/44)
* try-community-universal-mac-macosx10.9: [success](https://vulcan.livecode.com/builders/try-community-universal-mac-macosx10.9/builds/2338)
* try-community-x86-linux-debian8: [success](https://vulcan.livecode.com/builders/try-community-x86-linux-debian8/builds/2326)
* try-community-x86_64-linux-debian8: [success](https://vulcan.livecode.com/builders/try-community-x86_64-linux-debian8/builds/2325)
* try-community-x86-win32: [success](https://vulcan.livecode.com/builders/try-community-x86-win32/builds/2360)
* try-community-x86_64-win32: [success](https://vulcan.livecode.com/builders/try-community-x86_64-win32/builds/2300)

File: ide-support/revsaveasiosstandalone.livecodescript [hunk 2]
```diff
@@ -938,53 +938,60 @@ private command revDecorateImageAssets pTarget, pSettings, pBaseFolder, pAppBund
...
+         end if
+         
+         local tExtension
+         set the itemdelimiter to "."
+         put the last item of tFile into tExtension
+         
+         local tFileName
+         put format("livecode_launch_image@%s.%s", \
+               tScale, \
+               tExtension) into tFileName
```
Comment (livecodeali) [inline ide-support/revsaveasiosstandalone.livecodescript:962]: Does this need parameterisation by appearance?

Comment (montegoulding) [reply ide-support/revsaveasiosstandalone.livecodescript:962]: Hmm yes but oddly it was functioning correctly in testing

**Pull request merged**